packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1643,7 +1643,11 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       }
     }
 
-    _formatAndSetValue(value);
+    if (_isSelectionOnlyChange(value)) {
+      _handleSelectionChanged(value.selection, renderEditable, SelectionChangedCause.keyboard);
+    } else {
+      _formatAndSetValue(value);
```

Comment:
This also attempt to sync the remove editing value in the embedding side, I think we should still need to do that if it is a selection change? cc @gspencergoog .

Reply:
What is a "remove editing value"? Shouldn't it send a `TextEditingValue` with an empty `text` in that case? And that would fail the `_isSelectionOnlyChange` check.

Reply:
@chunhtai can you explain what you mean again?  I'm not sure I understand exactly what you're referring to.

Reply:
sorry it is a typo, i meant the remote editing value. The formatAndSetValue also called _updateRemoteEditingValueIfNeeded
https://github.com/flutter/flutter/blob/9e315985d0788ee04df48559896be46225ecf2f5/packages/flutter/lib/src/widgets/editable_text.dart#L1804

This will attempt to sync the current editing value to the embedding. I was wondering if selection only change needs to be sync to the embedding as well.

Reply:
Yes, it does, because otherwise the next editing operation won't operate on the same selection in both places.

Reply:
After receiving a new value from embedding, if it's a selection-only change, we won't be formatting the text. Which means that `_value` and `_receivedRemoteTextEditingValue` are certainly the same. Is there something I'm missing?

Reply:
If I understand correctly, the only reason we may want to update remote editing value is when the text formatting causes changes in the editing value, so we send it to the embedding to avoid getting out-of-sync.

Reply:
I took a look at the code in engine https://github.com/flutter/engine/blob/f7c7b41f0a0c72163d749321d070880fcac6402d/shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java#L577
It looks like we care about the selection value. I was under the impression that the engine should sync its own editing value before sends the updateEditingValue to the framework, but i couldn't find the code that does that. So It seems like we do need to call _updateRemoteEditingValueIfNeeded somewhere even if it is a selection only change?

Reply:
The `TextEditingValue` includes the current selection.  So a "selection only" change does cause a change to `_value` or `_receivedRemoteTextEditingValue`, which then needs to be synced.

Reply:
Maybe we're all in agreement here that it needs to be synced? I somehow feel like we're all saying the same thing...

Reply:
ah I see, the updateEditingState is send with the value inside the embedding, so it is already the same. yes, we don't need to sync it in this case.

Reply:
Ok, let me try to clarify a bit. I think we are using the word "sync" which has many meanings in this context.

On the framework side, we keep two variables:
- `_value`: The current editing value on the framework side.
- `_receivedRemoteTextEditingValue`: This is what the framework thinks is the current editing value on the engine side.

Whenever the editing value changes in the engine, it sends an `updateEditingValue` message to the framework. The framework then sets `_receivedRemoteTextEditingValue` to the new value received from the engine (this is one meaning of "sync"). Then the framework tries to format the text of the new editing value received from the engine. If the formatting causes changes to the editing value (i.e. `_value` != `_receivedRemoteTextEditingValue`) then the framework sends the new (formatted) `_value` to the engine via `_updateRemoteEditingValueIfNeeded` (another meaning of "sync").

In the case of this PR, a selection-only change means there'll be no formatting changes, hence `_value` and `_receivedRemoteTextEditingValue` are guaranteed to be the same. So there's no need to call `_updateRemoteEditingValueIfNeeded` to "sync" the editing value back to the engine.

Reply:
Thanks for the clarification. Makes sense now. LGTM.